src/experimental.js

```diff
@@ -0,0 +1,106 @@
+// @flow
+import Config from "react-native-config";
+import { AsyncStorage } from "react-native";
+import {
+  setEnvUnsafe,
+  isEnvDefault,
+  changes,
+} from "@ledgerhq/live-common/lib/env";
+import type { EnvName } from "@ledgerhq/live-common/lib/env";
+
+import logger from "./logger";
+
+export type FeatureCommon = {
+  name: EnvName,
+  title: string,
+  description: string,
+  shadow?: boolean,
+};
+
+export type FeatureToggle = {
+  type: "toggle",
+  valueOn?: any,
+  valueOff?: any,
+};
+
+export type Feature = FeatureCommon & FeatureToggle;
+
+export const experimentalFeatures: Feature[] = [
+  {
+    type: "toggle",
+    name: "MANAGER_DEV_MODE",
+    title: "Dev mode",
+    description: "enables developer apps in manager",
+  },
+  // {
+  //   type: "toggle",
+  //   name: "EXPERIMENTAL_USB",
+  //   title: "Experimental USB",
+  //   description:
+  //     "Alternative USB implementation that might help solving USB issues. Enabling this feature might create UI glitches.",
+  // },
+  // {
+  //   shadow: true,
+  //   type: "toggle",
+  //   name: "EXPERIMENTAL_EXPLORERS",
+  //   title: "Experimental explorers",
+  //   description: "switch to the new version of explorers",
+  // },
+  // {
+  //   shadow: true,
+  //   type: "toggle",
+  //   name: "FORCE_PROVIDER",
+  //   valueOn: 4,
+  //   valueOff: 1,
+  //   title: "Manager test app provider=4",
+  //   description: "enables yet `unreleased` apps in manager",
+  // },
+];
+
+const storageKey = "experimentalFlags";
+
+export const getStorageEnv = async () => {
+  try {
+    const maybeData = await AsyncStorage.getItem(storageKey);
+    return maybeData ? JSON.parse(maybeData) : {};
+  } catch (error) {
+    logger.critical(error);
+    return {};
+  }
+};
+
+export const setStorageEnvs = async (key: EnvName, val: string) => {
+  try {
+    const envs = await getStorageEnv();
+    envs[key] = val;
+    await AsyncStorage.setItem(storageKey, JSON.stringify(envs));
+  } catch (error) {
+    logger.critical(error);
+  }
+};
+
+export const isReadOnly = (key: EnvName) => key in Config;
+
+export const enabledExperimentalFeatures = (): string[] =>
+  // $FlowFixMe
+  experimentalFeatures.map(e => e.name).filter(k => isEnvDefault(k));
+
+(async () => {
+  const envs = await getStorageEnv();
+
+  /* eslint-disable guard-for-in */
+  for (const k in envs) {
+    setEnvUnsafe(k, envs[k]);
+  }
+
+  for (const k in Config) {
+    setEnvUnsafe(k, Config[k]);
+  }
+  /* eslint-enable guard-for-in */
+
+  changes.subscribe(async ({ name, value }) => {
+    if (experimentalFeatures.find(f => f.name === name) && !isReadOnly(name)) {
+      await setStorageEnvs(name, value);
```

Comment:
there are potential race condition here, an event can comes in the stream before the previous one to be awaited, making the first set failing because you getStorageEnv of the old envs on the second call.

a solution could be to use a `concatMap` and returning a `from(promise)` , but basically the current code being async/await does not solve the issue and it's the same as not awaiting.

I would suggest you just have a `setStorageEnvs()` without params that will just save from what is in the env (but i guess only the subset that is here?)